script/github-actions/create-cross-app-import-graph.js

```diff
@@ -0,0 +1,281 @@
+/* eslint-disable no-param-reassign */
+const fs = require('fs');
+const path = require('path');
+const findImports = require('find-imports');
+const core = require('@actions/core');
+
+function getImports(filePath) {
+  return findImports(filePath, {
+    absoluteImports: true,
+    relativeImports: true,
+    packageImports: false,
+  });
+}
+
+function diffIncludesSrcApplicationsFiles(diff) {
+  return diff.includes('diff --git a/src/applications');
+}
+
+function getAppNameFromFilePath(filePath) {
+  return filePath.split('/')[2];
+}
+
+function getAppPathFromFileDiff(fileDiff) {
+  const str = fileDiff.replace('diff --git a/', '');
+  return str.slice(0, str.indexOf(' '));
+}
+
+function sliceDiffIntoDiffForEachChangedFile(diff) {
+  const diffForEachChangedFile = [];
+  const startOfChange = 'diff --git a/src/applications';
+  const endOfChange = 'diff --git a/';
+  let startOfDiffIndex = null;
+
+  for (let i = 0; i <= diff.length; i += 1) {
```

Comment:
Would it make more sense to use `diff.forEach`? There are also a few others areas in this file.

Reply:
No, because I `return` out of the loop if a condition is met. I can't do that in a `forEach` loop; `return` simply goes to the next iteration in a `forEach` loop.

Reply:
Actually, yes, I can change this to a `forEach`. I can't change the others for the reasons mentioned above.

Reply:
I just went to change it to a `forEach` loop and realized why I used `for` -- it's because I need to sort of back track here, on line 48:

```js
    ) {
      diffForEachChangedFile.push(diff.slice(startOfDiffIndex, i));
      i -= 1; // reduce i by 1 so the next iteration picks up the beginning of this diff
      startOfDiffIndex = null;
```